### PR TITLE
chore(widget-chat): merge master into widget-dev

### PR DESCRIPTION
**Pull request by adamweeks**
A lot of the tests failed after the merge, so I used the spark mock from the web client and implemented it in our state tests.

### CODE REVIEW

**Comment by adamweeks** [comment]
This is the commit that has all of the test changes:
https://github.com/ciscospark/spark-js-sdk/pull/341/commits/2fdb516d09649809908b85050dd5671b0c0589d1

File: packages/widget-chat/src/modules/redux-spark/actions.test.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /* eslint max-nested-callbacks: ["error", 3] */
+jest.mock(`./spark`);
+const {createSpark} = require(`./spark`);
```
Comment (adamweeks) [inline packages/widget-chat/src/modules/redux-spark/actions.test.js]: Jest says that the babel-jest plugin will do this correctly and not hoist the import over the mock. I couldn't get it to work, so I'm manually requiring.

File: packages/widget-chat/src/modules/redux-spark/spark.test.js [hunk 2]
```diff
@@ -1,8 +0,0 @@
-import {createSpark} from './spark';
```
Comment (adamweeks) [inline packages/widget-chat/src/modules/redux-spark/spark.test.js]: Importing the actual spark added ~30 seconds to the test framework. We could probably mock out all of the plugins and test if we want to get full coverage. For now, I've removed the basic test.

File: packages/widget-chat/src/modules/redux-spark/spark.js [hunk 3]
```diff
@@ -3,6 +3,8 @@
 import '@ciscospark/plugin-avatar';
 import '@ciscospark/plugin-conversation';
 import '@ciscospark/plugin-flag';
+import '@ciscospark/plugin-mercury';
+import '@ciscospark/plugin-wdm';
```
Comment (bzang) [inline packages/widget-chat/src/modules/redux-spark/spark.js]: Did we actually need to add these imports to get the tests working? If not, we may want to ditch them.

Comment (adamweeks) [inline packages/widget-chat/src/modules/redux-spark/spark.js]: We didn't need them, but we are using them. Is there any cost/benefit to not having them? @ianwremmel?

Comment (adamweeks) [inline packages/widget-chat/src/modules/redux-spark/spark.js]: Removed them for this round. Will merge after CI

**Pull request merged**